core/src/main/scala/package.scala

```diff
@@ -40,4 +40,15 @@ package object sfs extends sfs.api.Api {
       case Failure(t) => l(t)
     }
   }
+
+  trait Functor[F[_]] {
+    def map[A, B](f: A => B): F[A] => F[B]
+  }
+  object Functor {
+    implicit def forMappable[A, M[A] <: { def map[B](f: A => B): M[B] }]: Functor[M] =
```

Comment:
In Scala code it's quite unusual to use structural types. For this use I however think it's vital because I hate to type an instance of `Functor` for every thing that has a `map` method.

This of course fails on the built-in classes which have that `CanBuildFrom` interface. We could add an implicit for `TraversableOnce` if needed. But I imagine we will move to psp-std at some point.

I don't know the impact on runtime performance of structural types, but you probably have a clear indication (and maybe an idea to fix that if needed).